x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/node_allocation/node_allocation.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiSelect,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiSpacer,
+  EuiLoadingSpinner,
+  EuiButton,
+} from '@elastic/eui';
+
+import { PHASE_NODE_ATTRS } from '../../../../constants';
+import { LearnMoreLink } from '../../../components/learn_more_link';
+import { ErrableFormRow } from '../../form_errors';
+import { useLoadNodes } from '../../../../services/api';
+
+interface Props {
+  phase: string;
+  setPhaseData: (dataKey: string, value: any) => void;
+  showNodeDetailsFlyout: (nodeAttrs: any) => void;
+  errors: any;
+  phaseData: any;
+  isShowingErrors: boolean;
+}
+
+const learnMoreLink = (
+  <Fragment>
+    <EuiSpacer size="m" />
+    <LearnMoreLink
+      text={
+        <FormattedMessage
+          id="xpack.indexLifecycleMgmt.editPolicy.learnAboutShardAllocationLink"
+          defaultMessage="Learn about shard allocation"
+        />
+      }
+      docPath="modules-cluster.html#cluster-shard-allocation-settings"
+    />
+  </Fragment>
+);
+
+export const NodeAllocation: React.FunctionComponent<Props> = ({
+  phase,
+  setPhaseData,
+  showNodeDetailsFlyout,
+  errors,
+  phaseData,
+  isShowingErrors,
+}) => {
+  const { isLoading, data: nodes, error, sendRequest } = useLoadNodes();
+
+  if (isLoading) {
+    return (
+      <Fragment>
+        <EuiLoadingSpinner size="xl" />
+        <EuiSpacer size="m" />
+      </Fragment>
+    );
+  }
+
+  if (error) {
+    const { statusCode, message } = error;
+    return (
+      <Fragment>
+        <EuiCallOut
+          style={{ maxWidth: 400 }}
+          title={
+            <FormattedMessage
+              id="xpack.indexLifecycleMgmt.editPolicy.nodeAttributesLoadingFailedTitle"
+              defaultMessage="Unable to load node attributes."
```

Comment:
Suggest removing the period here, but that's totally minor. If having a period at the end matches other Kibana error message conventions, then please ignore.